numpy/core/src/umath/ufunc_object.c

```diff
@@ -2538,7 +2715,7 @@ PyUFunc_GeneralizedFunction(PyUFuncObject *ufunc,
      */
     core_dim_ixs_size = 0;
     for (i = 0; i < nop; ++i) {
-        core_dim_ixs_size += core_num_dims[i];
+        core_dim_ixs_size += ufunc->core_num_dims[i];
```

Comment:
Should this be precalclulated and stored as part of the ufunc struct?

Reply:
I wondered about that, but it seems a bit excessive. Another thing one might do is to store it at the end of `core_offsets`, so that that really equal `cumsum(core_num_dims)`

Reply:
p.s. If we want to go this route, I do think this should be a different PR -- as long as we don't have a release, we are free to change the struct even without a version number change. And my overall sense is that it is rarely really needed: even now, one can just do
```
core_dim_ixs_size = ufunc->core_offsets[ufunc->nargs - 1] + ufunc->core_num_dims[ufunc->nargs - 1]
```